Source/Core/Core/AchievementManager.cpp

```diff
@@ -270,6 +272,75 @@ void AchievementManager::ActivateDeactivateRichPresence()
       nullptr, 0);
 }
 
+void AchievementManager::FetchBadges()
+{
+  if (!m_is_runtime_initialized || !IsLoggedIn() || !Config::Get(Config::RA_BADGES_ENABLED))
+    return;
+  if (!m_player_badge.loaded)
+  {
+    m_image_queue.EmplaceItem([this] {
+      {
+        std::lock_guard lg{m_lock};
+        std::string username = Config::Get(Config::RA_USERNAME);
+        rc_api_fetch_image_request_t icon_request = {.image_name = username.c_str(),
+                                                     .image_type = RC_IMAGE_TYPE_USER};
+        RequestImage(icon_request, &m_player_badge);
```

Comment:
These requests don't do anything if they fail.  Should they?

Reply:
Not sure there's a need to. If a badge load fails, the loaded flag is never set, and the code continues as if a badge just isn't loaded.

Reply:
So it just doesn't display the badge?  What if the user knows they have the badge but doesn't see it?  Would it be beneficial to display a log for diagnostic purposes?

Reply:
Hmmm. If the badge exists, the only reason I can think of why it would fail is an internet connection problem. Right now I don't really have anything in here for re-attempts if there's a connection failure, there was the possibility of that being future work but it hasn't been developed yet.

Reply:
As for logs, as I mentioned elsewhere I haven't done any kind of logging in RetroAchievements work yet. If we want to add proper error logging, I can do so, probably would need to go back and add it to several places it ought to be in code already checked in. Presumably we have other things being logged, I can look into that and work on a widescale logging update along with logs for the work currently in progress.